File: Week_07/G20200343030436/231.2-的幂.java
/*
 * @lc app=leetcode.cn id=231 lang=java
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution {
    public boolean isPowerOfTwo(int n) {
        if (n == 0) {
            return false;
        }
        long x = (long)n;
        return (x & (x - 1)) == 0;
    }
}

//class Solution {
    // func isPowerOfTwo(_ n: Int) -> Bool {
    //     if (n == 0) {
    //         return false
    //     }
    //     let x = n
    //     return (x & (x - 1)) == 0
    // }
//}
// @lc code=end

File: Week_07/G20200447010018/231.2-的幂.js
/*
 * @lc app=leetcode.cn id=231 lang=javascript
 *
 * [231] 2的幂
 *
 * https://leetcode-cn.com/problems/power-of-two/description/
 *
 * algorithms
 * Easy (47.55%)
 * Likes:    171
 * Dislikes: 0
 * Total Accepted:    50.6K
 * Total Submissions: 105.8K
 * Testcase Example:  '1'
 *
 * 给定一个整数，编写一个函数来判断它是否是 2 的幂次方。
 * 
 * 示例 1:
 * 
 * 输入: 1
 * 输出: true
 * 解释: 2^0 = 1
 * 
 * 示例 2:
 * 
 * 输入: 16
 * 输出: true
 * 解释: 2^4 = 16
 * 
 * 示例 3:
 * 
 * 输入: 218
 * 输出: false
 * 
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    while (n > 1) {
        if (n % 2) {
            break;
        } else {
            n = n / 2;
        }
    }
    return n === 1;
};
// @lc code=end

